### PR TITLE
Close #25 available spaces

### LINKED ISSUE #25: users can see available spaces only!
As a User,
So that I can choose a property that is available,
I want to filter the spaces by available dates.

### PR DESCRIPTION
**Pull request by kylebuttner**

### CODE REVIEW

File: app/views/spaces/index.erb [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
-    Description: <%= space.description %><br>
-    Price: <%= "£%.2f" % space.price %><br>
-    Available: <%= space.available_date %><br>
-    Username: <%=space.user.username %><br>
-    <form action="/bookings/new" method="post">
-      <input type="hidden" name="space_id" value="<%= space.id %>">
-      <input type="submit" value="Book now">
-    </form>
-  </li>
+  <form action='/spaces' method="get">
```
Comment (anne-lydon) [inline app/views/spaces/index.erb]: use double quotes in HTML (also to be consistent here)

File: app/views/spaces/index.erb [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
+    Displaying Spaces available on <%= @display_date %>, select another date:<br>
+    <input type='date' name='filter_date'>
+    <input type='submit' name='submit' value='Filter Date'>
```
Comment (anne-lydon) [inline app/views/spaces/index.erb]: can we use a button tag to be more semantic?

File: spec/features/web_helper.rb [hunk 3]
```diff
@@ -43,3 +44,17 @@ def create_booking
   fill_in :date_requested, with: '01/01/2016'
   click_button 'Submit'
 end
+
+#all db_create functions expect you to pass data in the right format (date in Date, user as an User obj...)
```
Comment (anne-lydon) [inline spec/features/web_helper.rb]: remove comment

File: app/controller/spaces.rb [hunk 4]
```diff
@@ -3,8 +3,13 @@ class Bnb < Sinatra::Base
   get '/spaces' do
 
    redirect('/') unless User.get(session[:user_id])
-
+   if params[:filter_date]
+    @display_date = Date.parse(params[:filter_date])
```
Comment (anne-lydon) [inline app/controller/spaces.rb]: Can we change the `@display_date` to `@filter_date` to make it clear?

File: app/views/spaces/index.erb [hunk 5]
```diff
@@ -1,17 +1,27 @@
...
+  </form>
+  <% @spaces.each do |space|
+        space_bookings = @bookings.all(space_id: space.id) || []
+        not_booked = space_bookings.all? { |booking| booking.status!="confirmed" }
```
Comment (anne-lydon) [inline app/views/spaces/index.erb]: can we change `not_booked` to `not_confirmed` ?

File: app/views/spaces/index.erb [hunk 6]
```diff
@@ -1,17 +1,27 @@
...
+        listed = @display_date ? space.available_date == @display_date : true
+        if not_booked && listed %>
```
Comment (AnnemarieKohler) [inline app/views/spaces/index.erb]: Should we get rid of `listed` altogether, as it won't ever be false anyway?

**Comment by anne-lydon** [comment]
:shipit:

File: app/views/spaces/index.erb (hunk 1 above)
Comment (kylebuttner) [inline app/views/spaces/index.erb]: done!

File: app/views/spaces/index.erb (hunk 2 above)
Comment (kylebuttner) [inline app/views/spaces/index.erb]: done!

File: spec/features/web_helper.rb (hunk 3 above)
Comment (kylebuttner) [inline spec/features/web_helper.rb]: done!

File: app/controller/spaces.rb (hunk 4 above)
Comment (kylebuttner) [inline app/controller/spaces.rb]: done!

File: app/views/spaces/index.erb (hunk 5 above)
Comment (kylebuttner) [inline app/views/spaces/index.erb]: done! and changed `listed` to `available` for clarity.

**Comment by kylebuttner** [comment]
@AnnemarieKohler @ccfz adjustments complete

**Comment by robota-x** [comment]
:shipit:

**Pull request merged**